web/satellite/src/components/dialogs/BucketDetailsDialog.vue

```diff
@@ -119,26 +120,43 @@ const showRegionTag = computed<boolean>(() => {
     return configStore.state.config.enableRegionTag;
 });
 
-function redirectToBucketsPage(): void {
-    router.push({
-        name: ROUTES.Buckets.name,
-        params: { id: projectsStore.state.selectedProject.urlId },
-    });
-}
-
-const bucket = computed((): Bucket => {
-    if (!projectsStore.state.selectedProject.id) return new Bucket();
+/**
+ * Fetch the bucket data if it's not avaliable
+ */
+async function loadBucketData() {
+    if (!projectsStore.state.selectedProject.id) {
+        bucket.value = new Bucket();
+        return;
+    }
 
     const data = bucketsStore.state.page.buckets.find(
         (bucket: Bucket) => bucket.name === props.bucketName,
     );
 
-    if (!data) {
-        redirectToBucketsPage();
-
-        return new Bucket();
+    if (data) {
+        bucket.value = data;
+    } else {
+        try {
+            const fetchedBucket = await bucketsStore.getSingleBucket(
+                projectsStore.state.selectedProject.id,
+                props.bucketName,
+            );
+
+            bucket.value = fetchedBucket;
+        } catch (error) {
+            console.error('Failed to fetch bucket data', error);
```

Comment:
Ok, instead of using console methods, we have a notification system. You could do something like this so the user can see what's wrong.
```
notify.notifyError(error, AnalyticsErrorEventSource.BUCKET_DETAILS_MODAL);
```

you'd have to import `useNotify` composable and add `BUCKET_DETAILS_MODAL` constant here
https://github.com/storj/storj/blob/main/web/satellite/src/utils/constants/analyticsEventNames.ts#L57-L121

something like this
```
BUCKET_DETAILS_MODAL = 'Bucket details modal',
```